lib/collection/src/collection_manager/holders/proxy_segment.rs

```diff
@@ -473,23 +465,39 @@ impl SegmentEntry for ProxySegment {
     }
 
     fn estimate_points_count<'a>(&'a self, filter: Option<&'a Filter>) -> CardinalityEstimation {
-        let deleted_points_count = self.deleted_points_count.load(Ordering::Relaxed);
-        let wrapped_segment_count = self
-            .wrapped_segment
+        let deleted_points_count = self.deleted_points.read().len();
+
+        let (wrapped_segment_count, total_wrapped_size) = {
+            let wrapped_segment = self.wrapped_segment.get();
+            let wrapped_segment_guard = wrapped_segment.read();
+            (
+                wrapped_segment_guard.estimate_points_count(filter),
+                wrapped_segment_guard.points_count(),
+            )
+        };
+
+        let write_segment_count = self
+            .write_segment
             .get()
             .read()
             .estimate_points_count(filter);
+
+        let expected_deleted_count = if total_wrapped_size > 0 {
+            (wrapped_segment_count.exp as f64
+                * (deleted_points_count as f64 / total_wrapped_size as f64)) as usize
+        } else {
+            0
+        };
+
         CardinalityEstimation {
             primary_clauses: vec![],
             min: wrapped_segment_count
```

Comment:
Why not like in exp?
```
min: (wrapped_segment_count.min + write_segment_count.min)
                .saturating_sub(deleted_points_count),
```
I guess it is possible to get `exp < min` in current version. But I'm not sure for now

Reply:
Oh, I see that `expected_deleted_count` cannot be more than `wrapped_segment_count.exp` - we don't need `saturating_sub ` and my first comment is obsolete